registry/storage/catalog.go

```diff
@@ -93,5 +99,38 @@ func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error)
 		}
 	}
 	return nil
+}
+
+// todo(richardscothern): find a better place for this
+func (reg *metadataRegistry) Repositories(ctx context.Context, repos []string, lastRepo string) (n int, errVal error) {
+	if len(repos) == 0 {
+		return 0, errors.New("no space in slice")
+	}
+
+	var items []string
+
+	params := metadata.IterateParameters{From: lastRepo, IterType: metadata.RepoKey{}}
+	err := reg.metadataService.Iterate(ctx, params, func(key string, val interface{}) error {
+		items = append(items, key)
+		if len(items) >= len(repos) {
+			// filled the give slice
+			return ErrFinishedWalk
```

Comment:
Iterate is done when the given slice is filled (ErrFinishedWalk) or when the catalog is enumerated (io.EOF).  There difference is to let the client know it should keep trying to get more.